### PR TITLE
Small changes to contrib/mkimage-yum.sh

### PR DESCRIPTION
**Pull request by FlorinAsavoaie**
Added -g and -p options for the contrib/mkimage-yum.sh script to allow generating images with the dependencies you want. I use this because I generate images where I only need "glibc" with its dependencies or "java-1.8.0-openjdk-headless" and so on. This makes sure that I don't get a lot of bogus packages from the Core group.

Possible usages now include `sudo ./mkimage-yum.sh -y yum.conf -g -p java-1.8.0-openjdk-headless`.

Also changed the "test" at the end to add --rm parameter to docker run so it doesn't leave a junk container and use bash to echo "success" in case you don't get coreutils in your dependencies tree.

### CODE REVIEW

**Comment by thaJeztah** [comment]
Thanks!

ping @tianon @jfrazelle for review

File: contrib/mkimage-yum.sh [hunk 1]
```diff
@@ -56,13 +67,24 @@ mknod -m 666 "$target"/dev/urandom c 1 9
 mknod -m 666 "$target"/dev/zero c 1 5
 
 # amazon linux yum will fail without vars set
-if [ -d /etc/yum/vars ]; then
-	mkdir -p -m 755 "$target"/etc/yum
-	cp -a /etc/yum/vars "$target"/etc/yum/
+if [ -d /etc/yum/vars ];
```
Comment (jfrazelle) [inline contrib/mkimage-yum.sh]: these are just formatting? is there anything its fixinng?

Comment (FlorinAsavoaie) [inline contrib/mkimage-yum.sh]: Nope, I just fixed the formatting so it is consistent across the file.

Comment (thaJeztah) [inline contrib/mkimage-yum.sh]: For a cleaner history, might be better to leave that for a separate PR, but I'll leave that up to @jfrazelle

**Comment by FlorinAsavoaie** [comment]
I found a bug in the code I submitted. I will fix it and open another PR. Sorry

**Comment by thaJeztah** [comment]
@FlorinAsavoaie once fixed, you can also re-open this issue and push to GitHub, that will update this PR; no need to open a new PR in that case

**Pull request closed without merging**